src/services/resources-getter.js

```diff
@@ -223,13 +223,13 @@ function ResourcesGetter(model, opts, params) {
 
   function getSegmentCondition() {
     getSegment();
-    if (_.isFunction(segmentWhere)) {
+    if (isFunction(segmentWhere)) {
       return segmentWhere(params)
         .then((where) => {
           segmentWhere = where;
         });
     }
-    return P.resolve();
+    return _resolve();
```

Comment:
I think I get why you did that but I find it a bit ugly/hacky to read.

Reply:
Why don't we keep `P.resolve()`? I find it easier to read because we know that `P` is `bluebird` dependency.
Here if I want to know where does this function comes, I have to find in the file where it comes from.

I have the same opinion about:
- `_.find`
- `_.union`
- `_.each`
... (at least all the things that comes from external dependencies)